src/tools/compiletest/src/header.rs

```diff
@@ -395,7 +402,15 @@ impl TestProps {
                 );
                 config.set_name_directive(ln, STDERR_PER_BITWIDTH, &mut self.stderr_per_bitwidth);
                 config.set_name_directive(ln, INCREMENTAL, &mut self.incremental);
-                config.set_name_directive(ln, KNOWN_BUG, &mut self.known_bug);
+                config.set_name_value_directive(ln, KNOWN_BUG, &mut self.known_bug, |b| {
+                    match b.trim() {
+                        "unknown" => KnownBug::Unknown,
+                        bug if Regex::new(r".*#\d+").unwrap().is_match(bug) => {
+                            KnownBug::Ref(bug.to_string())
```

Comment:
It doesn't really matter, but it seems like there's no advantage in storing this after parsing -- can we keep the known_bug field as a bool?

Reply:
Done.